architecture/2024-08-controller-component-threat-model.md

```diff
@@ -0,0 +1,241 @@
+
+3. **Controller Reconciliation**: When the action (create/update/delete) is validated by the Validating Webhook (applicable only to the Recipe resource and not to the Deployment resource) as discussed in the previous step, the Kubernetes controllers we have (Recipe and Deployment Reconcilers) watch for changes via the Kubernetes API Server. The controllers then reconcile the resource's state to match the desired configuration specified in the resource definition. This involves ensuring that the actual state of the cluster matches the desired state, making any necessary adjustments to achieve this.
+
+4. **Communication with UCPD**: Controller also calls UCPD when doing reconcilations. Here is the list of instances where Controller calls UCPD:
+
+   1. Recipe Reconciler:
+
+      - Poll operations like create, update, or delete a Recipe resource.
+      - Resolve dependencies for a Recipe resource:
+        - Get the Radius environment the Recipe is associated with.
+        - Create the Radius Resource Group if needed.
+        - Create the Radius Application if needed.
+      - Fetch a Radius resource.
+      - List Secrets of a Recipe resource.
+      - Delete a Recipe resource.
+
+   2. Deployment Reconciler:
+
+      - Poll operations like create, update, or delete a Deployment resource.
+      - Resolve dependencies for a Deployment resource:
+        - Get the Radius environment the Deployment is associated with.
+        - Create the Radius Resource Group if needed.
+        - Create the Radius Application if needed.
+      - Fetch a Recipe resource.
+      - List Secrets of a Recipe resource.
+
+5. **Communication with Kubernetes API Server**: Controller also calls the Kubernetes API Server when doing reconcilations. Here is the list of instances where Controller calls the Kubernetes API Server:
+
+   1. Recipe Reconciler:
+
+      - Fetch a Recipe object.
+      - Send events related to the operations.
+      - Update a Recipe object:
+        - Status update.
+        - Addition and removal of a Finalizer.
+      - Delete a Secret object that is associated with the Recipe.
+      - Create a Secret object to associate with the Recipe if needed.
+
+   2. Deployment Reconciler:
+
+      - Fetch a Deployment object.
+      - Send events related to the operations.
+      - Update a Deployment object:
+        - Status update.
+        - Addition and removal of a Finalizer.
+      - Delete a Secret object associated with a Deployment that is no longer needed.
+      - Fetch a Secret object.
+      - Create or Update a Secret object and associate it with a Deployment object.
+      - List Deployments filtered with specific Recipe objects.
+
+### Threats
+
+#### Impersonation of Radius Controllers to trigger resource access
```

Comment:
Access to the Radius API:
- everybody (admin users) can send requests to the Radius API

What are all the possible clients of Radius API? We own and other users own.

RBAC can solve this issue instead of Network Policies

mTLS, Network Policies, RBAC